### DISCUSSION TITLE
Links open in iframe when using response caching with inertia

### DISCUSSION THREAD

**Discussion by fh32000** [spatie/laravel-responsecache#448] [Bugs]
I'm using the spatie/laravel-responsecache library to cache my responses, and when I enable caching, the links open in an iframe. When I disable caching, the links work as expected.

I've tried disabling the iframe in the inertia.config.js file, but that didn't work.

I'm using Laravel 10.15.0 and Vue 3.2.36 vite 3.2.2 vuetify 3.0 @inertiajs/inertia 0.11.1

Here is a code snippet that reproduces the issue:
```
<php
Route::get('home', HomeController::class)->name('home')
    ->middleware('cacheResponse:3600');
});
```

When I visit this route, the links open in an iframe.

I'm hoping you can help me figure out what's causing this issue.

see -> https://github.com/inertiajs/inertia/issues/586#issuecomment-1691820008

Thanks,
@fh32000

**Comment by fh32000** [comment]
@ArondeParon @intrepidws @willoller  any updates

**Comment by jonerickson** [comment]
Dealing with this as well. Any leads?

**Comment by MaherRemita** [comment]
I was able to find a solution for this thanks to SabatinoMasala from the Sabatino Talks Dev YouTube Channel 
[https://www.youtube.com/watch?v=U5eViAKHD0o](url) 
in this video, he explained the problem and the solution to it 

here is what I did step by step :

-  create a controller class that extends the **BaseCacheProfile** class 

- add the following code in the class 
`<?php

namespace App\Http\Controllers;

use Illuminate\Support\Str;
use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;
use Spatie\ResponseCache\CacheProfiles\BaseCacheProfile;

class InertiaResponse extends BaseCacheProfile
{
    public function useCacheNameSuffix(Request $request): string
    {
        return request()->inertia() ? 'inertia' : 'no-inertia';
    }

    public function shouldCacheRequest(Request $request): bool
    {
        // if ($request->ajax()) {
        //     return false;
        // }

        if ($this->isRunningInConsole()) {
            return false;
        }

        return $request->isMethod('get');
    }

    public function shouldCacheResponse(Response $response): bool
    {
        if (! $this->hasCacheableResponseCode($response)) {
            return false;
        }

        if (! $this->hasCacheableContentType($response)) {
            return false;
        }

        return true;
    }

    public function hasCacheableResponseCode(Response $response): bool
    {
        if ($response->isSuccessful()) {
            return true;
        }

        if ($response->isRedirection()) {
            return true;
        }

        return false;
    }

    public function hasCacheableContentType(Response $response): bool
    {
        $contentType = $response->headers->get('Content-Type', '');

        if (str_starts_with($contentType, 'text/')) {
            return true;
        }

        if (Str::contains($contentType, ['/json', '+json'])) {
            return true;
        }

        return false;
    }
}
`
- in the laravel response-cache config file replace :
`'cache_profile' => Spatie\ResponseCache\CacheProfiles\CacheAllSuccessfulGetRequests::class,`
with 
`'cache_profile' => App\Http\Controllers\InertiaResponse::class,` (or what ever the name of the class you created)

that's is now you should be good

**Reply by thecrazybob** [reply]
This worked for me as well. [Link to the original source](https://github.com/SabatinoMasala/filament-marketing-starter/blob/main/app/Http/CacheProfiles/InertiaAwareCacheProfile.php)
